dataland-frontend/src/components/forms/parts/kpiSelection/KPIfieldSet.vue

```diff
@@ -0,0 +1,166 @@
+        :explanation="kpiInfoMappings[name] ?? ''"
+      />
+      <FormKit
+        :disabled="!dataPointIsAvailable"
+        type="number"
+        name="value"
+        v-model="currentMainValue"
+        validation-label=""
+        :placeholder="valueType === 'percent' ? 'Value %' : 'Value'"
+        step="any"
+        min="0"
+        :validation="valueType === 'percent' ? 'number|between:0,100' : 'number'"
+        :inner-class="{
+          short: true,
+        }"
+      />
+    </div>
+
+    <div class="form-field">
+      <FormKit type="group" v-if="dataPointIsAvailable" name="dataSource">
+        <h4 class="mt-0">Data source</h4>
+        <div class="next-to-each-other">
+          <div class="flex-1">
+            <UploadFormHeader :name="kpiNameMappings.report ?? ''" :explanation="kpiInfoMappings.report ?? ''" />
+            <FormKit
+              type="select"
+              name="report"
+              v-model="currentReportValue"
+              :disabled="!dataPointIsAvailable"
+              placeholder="Select a report"
+              :options="['None...', ...reportsName]"
+            />
+          </div>
+          <div>
+            <UploadFormHeader :name="kpiNameMappings.page ?? ''" :explanation="kpiInfoMappings.page ?? ''" />
+            <FormKit
+              outer-class="w-100"
+              :disabled="!dataPointIsAvailable"
+              v-model="currentPageValue"
+              type="number"
+              name="page"
+              placeholder="Page"
+              validation-label="Page"
+            />
+          </div>
+        </div>
+      </FormKit>
+    </div>
+
+    <!-- Data quality -->
+    <div class="form-field">
+      <UploadFormHeader name="Data quality" explanation="Data quality" />
+      <div class="md:col-6 col-12 p-0">
+        <FormKit
+          :disabled="!dataPointIsAvailable"
+          type="select"
+          v-model="currentQualityValue"
+          name="quality"
+          :validation="dataPointIsAvailable ? 'required' : ''"
+          validation-label="Data quality"
+          placeholder="Data quality"
+          :options="dataQualityList"
+        />
+      </div>
+    </div>
+  </div>
+  <div class="form-field">
+    <FormKit
+      type="textarea"
+      name="comment"
+      placeholder="(Optional) Add comment that might help Quality Assurance to approve the datapoint. "
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import InputSwitch from "primevue/inputswitch";
+import UploadFormHeader from "@/components/forms/parts/UploadFormHeader.vue";
+import { FormKit } from "@formkit/vue";
+
+export default defineComponent({
+  name: "KPIfieldSet",
+  components: { UploadFormHeader, FormKit, InputSwitch },
+  emits: ["dataPointAvailableToggle"],
+  data: () => ({
+    dataPointIsAvailable: true,
+    dataQualityList: ["NA", "Audited", "Reported", "Estimated", "Incomplete"],
```

Comment:
I think that this should not be hardcoded, because it can be derived from our autogenerated models:
Check out `QualityOptions` in the build folder of `dataland-frontend`.

Reply:
ok

Reply:
Reminder for myself:  Check if options still displayed correctly in actual frontend

Reply:
The current code there could be simplified. Some suggestions:
- Write `qualityOptions` instead of `QualityOptions` because otherwise you could mistake it for the actual enum class!
- Why using `.entries()` if only the values are needed?

Reply:
Adjusted this and it works.